pkg/controller/jobs/mpijob/mpijob_controller.go

```diff
@@ -54,20 +51,16 @@ func init() {
 	}))
 }
 
-// MPIJobReconciler reconciles a Job object
-type MPIJobReconciler jobframework.JobReconciler
-
-func NewReconciler(
-	scheme *runtime.Scheme,
-	client client.Client,
-	record record.EventRecorder,
-	opts ...jobframework.Option) jobframework.JobReconcilerInterface {
-	return (*MPIJobReconciler)(jobframework.NewReconciler(scheme,
-		client,
-		record,
-		opts...,
-	))
-}
+// +kubebuilder:rbac:groups=scheduling.k8s.io,resources=priorityclasses,verbs=list;get;watch
+// +kubebuilder:rbac:groups="",resources=events,verbs=create;watch;update;patch
+// +kubebuilder:rbac:groups=kubeflow.org,resources=mpijobs,verbs=get;list;watch;update;patch
+// +kubebuilder:rbac:groups=kubeflow.org,resources=mpijobs/status,verbs=get;update
+// +kubebuilder:rbac:groups=kueue.x-k8s.io,resources=workloads,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=kueue.x-k8s.io,resources=workloads/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=kueue.x-k8s.io,resources=workloads/finalizers,verbs=update
+// +kubebuilder:rbac:groups=kueue.x-k8s.io,resources=resourceflavors,verbs=get;list;watch
+
+var NewReconciler = jobframework.NewGenericReconciler(func() *MPIJob { return &MPIJob{&kubeflow.MPIJob{}} }, nil)
```

Comment:
```suggestion
var NewReconciler = jobframework.NewGenericReconciler(func() jobframework.GenericJob { return &MPIJob{&kubeflow.MPIJob{}} }, nil)
```
if the other remark about unnecessary generics is applied.